packages/flutter/lib/src/material/tabs.dart

```diff
@@ -389,7 +391,10 @@ class _IndicatorPainter extends CustomPainter {
       tabRight -= delta;
     }
 
-    return Rect.fromLTWH(tabLeft, 0.0, tabRight - tabLeft, tabBarSize.height);
+    final EdgeInsets insets = indicatorPadding.resolve(_currentTextDirection);
+    final Rect rect = Rect.fromLTWH(tabLeft, 0.0, tabRight - tabLeft, tabBarSize.height);
+
+    return insets.deflateRect(rect);
```

Comment:
This need to handle where insets > original rect, it may become native rect

Reply:
> This need to handle where insets > original rect, it may become native rect
A few assert statements should fix this, i guess. I'll get to it.
>  i think the correct solution is not to clip even without the padding.
I was looking for a way to be able to do that, but I think the way this works is that TabBar draws a Canvas according to the space available to it, and then the indicator is being painted on that.(This is what my understanding of the code says, I might be wrong here.) Painting outside the canvas is obviously a no-no. Could there be a way to be able to increase the size of canvas? That seems like an odd way to solve this because, I think its hard to ever know how big it should be. What do you think about this? Please let me know.

Reply:
throw a flutter error makes more sense because this is a user define value.